hearpreprocess/pipeline.py

```diff
@@ -474,12 +486,51 @@ def run(self):
         metadata = self.get_all_metadata()
         print(f"metadata length = {len(metadata)}")
 
+        def _log_metadata_split_size_label(metadata, event_str):
+            diagnostics.info(
+                f"{self.longname} Files in each split {event_str}: "
+                "{}".format(metadata.groupby("split")["relpath"].nunique().to_dict())
+            )
+            diagnostics.info(
+                f"{self.longname} Unique labels in each split {event_str}: "
+                "{}".format(metadata.groupby("split")["label"].nunique().to_dict())
+            )
+
+        _log_metadata_split_size_label(metadata, "Before Postprocessing")
         metadata = self.postprocess_all_metadata(metadata)
+        _log_metadata_split_size_label(metadata, "After Postprocessing")
 
         # Split the metadata to create valid and test set from train if they are not
         # created explicitly in get_all_metadata
         metadata = self.split_train_test_val(metadata)
 
+        # Each split should have unique files and no file should be across splits
+        assert (
+            len(
+                set.intersection(
+                    *map(set, metadata.groupby("split")["relpath"].apply(list).tolist())
+                )
+            )
+            == 0
+        ), "Relpath is not unique across split"
+
+        _log_metadata_split_size_label(metadata, "After Splitting")
+
+        diagnostics.info(
+            f"{self.longname} Existing Filepaths in metadata  for each split: "
```

Comment:
I don't get it. Why do we want this? Where is the after diagnostic for this before diagnostic?

And it's formatted really ugly and verbose

"Existing files in original metadata:

Is that correct? It's hard to understand just by reading the log top to bottom what this actually is. The diagnostics should be self-explanatory and not expect you to be reading the code at the same time